lib/buffer.js

```diff
@@ -401,7 +400,8 @@ Buffer.isBuffer = function isBuffer(b) {
 
 Buffer.compare = function compare(a, b) {
   if (!isUint8Array(a) || !isUint8Array(b)) {
-    throw new TypeError('Arguments must be Buffers or Uint8Arrays');
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'arguments',
```

Comment:
This does not quite fit with the new approach.  It really should be one of 'a' or 'b' instead of 'arguments'.  I think we may need to different cases one that throws an error for 'a' and one for 'b' to be consistent with how we through errors everywhere else.  To match that the second string passed in should be the name of the argument defined in the function definition that is wrong.